### ISSUE TITLE
lint rule - unused call result

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3668]
one of my favorite pyright rules is `reportUnusedCallResult`, which warns when you may be misusing a function that returns a value by not actually using the result. for example:

Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&code=CYUwZgBAxg9gtgIwJYDsQG0AqBdAFEgFxDgGcAuCAGyRIK2wBoI0B3AfUOIswEoIBaAHxUadHGQBQEaRABOIAgFdZKCOgBUnUk1YcicbBKNaSEALxqAjAwBMhiQGIIcGopIgIMSAQAWH2IioHmCKKFAESDAoTL4ebiCyEL4wigDmPgQQhM6KBACGRKaxciAkipSZCLkQeeGKeZSUAJ4lSiqmecwgLCK0kgHIaPj6JEwAzHxOCbIwshQASqXlmV7QDZQQIAAeAA7yJCSRqjSe3k07HgBE1LToqATYlzUowFmmKDCZ8cAA3DUHSFSqgIMAgADc8rIkHkEJQrmwnkhvD4TqiUER0UcGkYJHt7sNiCRJp5KK8IZRFCAgA)

```python
def combine[T](items: list[T], new_item: T) -> list[T]:
    return [*items, new_item]


items = [1,2]

# misuse of the combine function, the user thought it mutates the result but actually returns a new list:
combine(items, 3) # error: Result of call expression is of type "list[int]" and is not used; assign to variable "_" if this is intentional


print(items) # old value
```

it would be great if ty had an option to warn against this. i realize that this is more of a lint rule (and one with quite a high chance for false positives) but i'm raising it here instead of ruff since it requires type information

**Comment by DetachHead** [comment]
`reportUnusedCoroutine` is another rule that does the same thing but specifically for coroutines.
